src/lib/SCEditor.js

```diff
@@ -1414,6 +1430,200 @@ define(function (require) {
 			});
 		};
 
+
+		/**
+		 * Adds a handle used to remap keys to make them more useful for code
+		 * input
+		 *
+		 * @private
+		 */
+		var addBodyCodeInputHandle = function (searchSelectors) {
+			$wysiwygBody.on('keydown', null,
+			{
+				'searchSelector': searchSelectors.join(',')
+			}, codeInputHandle);
+
+		};
+
+		/**
+		 * Adds a handle used to remap keys to make them more useful for
+		 * code input
+		 *
+		 * @private
+		 */
+		var codeInputHandle = function (e) {
+
+			if (e.ctrlKey || e.altKey || e.metaKey) {
+				return;
+			}
+
+			// For now, only Enter and Tab have defined executions
+			if (e.which !== 13 &&	// Enter
+				e.which !== 9		// tab
+			) {
+				return;
+			}
+
+			if (e.shiftKey && e.which !== 9) {
+				return;
+			}
+
+			rangeHelper.mergeTextNodesAtCaret();
+
+			var currentRange = rangeHelper.selectedRange();
+
+			if (!rangeHelper.getFirstBlockParent().
+				matches(e.data.searchSelector)) {
```

Comment:
Just noticed `matches()`  needs to be prefixed in IE (yes, IE really is a pain) and also for older Android and iOS: http://caniuse.com/#feat=matchesselector

It might be easier to use the jQuery `is()` instead since that should work everywhere.